Automattic/simplenote-ios
# Localize keyword for tag search

### Fix
This issue fixes #860 

When doing a tag search in Simplenote iOS, currently you can do so with the keyword tag: but not a localized version of the word tag.  Originally in issue 860, only the localized version worked, but through some changes to SNiOS the problem reversed, only "tag:" would work and not the localized version.  With this PR both tag: and localTag: will work.

<img width="300px;" src="https://cdn-std.droplr.net/files/acc_593859/DOrN20" />

### Test
1. After loading the branch, check and make sure that the correct [Simplenote Search branch](https://github.com/Automattic/SimplenoteSearch-Swift/tree/issue/860-localize-tag-search) has also been loaded
2. In the localizedString files, choose a language and add a key pair for "tag:" = _the localized version of tag: in the chosen language_ I used French which would be `"tag:" = "étiquette";`
3. Load up the simulator, with the simulator still set to English.  Go to a search and look for a tag by name It should appear with tag: as the prefix
4. While still in English type in tag: and then search for a tag by name, the prefix should be tag:
5. Switch the simulator to the language you added the translation for
6. Return to search and repeat steps 3 and 4 to make sure that none localized tag: still works
7. In the search field type in the localized version of the tag `étiquette:` or your chosen translation.  With the translated key complete in the search field the results will show the word tag translated

One thing to consider here, is it worth translating the result if a user enters étiquette: ?  Given how specific the entry has to be or that event to trigger, I am not sure that it is.  I had another possible option, where the key tag: does not appear in the search results.  Currently there is an icon that appears next to the tags and there is a header which is localized declaring the tag search results section.  We could remove tag:/étiquette: from the result and simply rely on the header and icon. 

One more thing to consider that I am not 100% sure on here.  One of the things that we had been looking for is to make sure that NSLocalizedString gets placed in such a way that the scripts can pick up the strings and then change them.  Because the SearchQuerySettings struct is in SimplenoteSearch, I am not sure that this will happen.  Can you take a look to see if the Strings should get picked up by the localizing scripts, and if not, can you advise on changes to make to fix it? 

### Review
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
